lib/SILOptimizer/Utils/ConstExpr.cpp

```diff
@@ -1370,6 +1370,23 @@ ConstExprFunctionState::evaluateFlowSensitive(SILInstruction *inst) {
     return None;
   }
 
+  if (isa<DestructureTupleInst>(inst) || isa<DestructureStructInst>(inst)) {
```

Comment:
I think you could probably do this in a more general way, but I think for now this is fine.